File: java/greeter/src/Greeter.java
import java.util.Scanner;

public class Greeter {
  public static class InputHandler {
    public static String getUserName() {
      Scanner scanner = new Scanner(System.in);
      System.out.print("Enter your name: ");
      String userinput = scanner.nextLine();

      String name = userinput.trim();

      return name;
    }
  }

  private String username;

  public Greeter(String name) {
    this.username = name;
  }

  public void greet() {
    System.out.println("Hello, " + this.username + "!");
    System.out.println("Greetings from Java!");
  }
  public static void main(String[] args) {
    String name = InputHandler.getUserName();
    Greeter greeter = new Greeter(name);

    greeter.greet();
  }
}

File: kotlin/greeter/src/main/kotlin/greeter.kt
package greeter

import java.util.Scanner

object InputHandler {
  fun getUserName(): String {
    val scanner = Scanner(System.`in`)
    print("Enter your name: ")
    var name = scanner.nextLine().trim()

    if (name.isEmpty()) {
      name = "Stranger"
    }

    return name
  }
}

class Greeter(private val username: String) {
  fun greet() {
    println("Hello, $username!")
    println("Greetings from Kotlin!")
  }
}

fun main() {
  val name = InputHandler.getUserName()
  val greeter = Greeter(name)

  greeter.greet()
}
